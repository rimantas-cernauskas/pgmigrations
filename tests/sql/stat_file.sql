-- Test function pgmigrations.stat_file()
\i tests/sql/setup/init.sql

/*

This test is relying on the existece of PG_VERSION file

*/

BEGIN;
SELECT plan(2);

PREPARE my_query AS
    SELECT * FROM pgmigrations.stat_file( $1 );

PREPARE my_expect AS
    SELECT * FROM PG_STAT_FILE( $1 );

SELECT results_eq(
     'EXECUTE my_query(''./'')'
    ,'EXECUTE my_expect(''./'')'
    ,'should return same result as PG_STAT_FILE for a directory'
);

SELECT results_eq(
     'EXECUTE my_query(''PG_VERSION'')'
    ,'EXECUTE my_expect(''PG_VERSION'')'
    ,'should return same result as PG_STAT_FILE for a file'
);

DEALLOCATE my_query;
DEALLOCATE my_expect;

SELECT * FROM finish();
ROLLBACK;