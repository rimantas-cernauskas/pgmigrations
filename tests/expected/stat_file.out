-- Test function pgmigrations.stat_file()
\i tests/sql/setup/init.sql
\set ECHO errors
1..2
ok 1 - should return same result as PG_STAT_FILE for a directory
ok 2 - should return same result as PG_STAT_FILE for a file
